pybop/plotting/plot2d.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import pybop
+import numpy as np
+
+
+def plot2d(
+    cost_or_optim, gradient=False, bounds=None, steps=10, show=True, **layout_kwargs
```

Comment:
Something like this could be a bit more clear. Could helpful with generality in the future as well.

```suggestion
    plot_object, gradient=False, bounds=None, steps=10, show=True, **layout_kwargs
```

Reply:
Let's update this when we allow for more general inputs in the future.